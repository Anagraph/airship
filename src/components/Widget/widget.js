import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import SubHeader from '../Typography/subheader';
import Text from '../Typography/text';
import { colors } from '../../constants';

const StyledWidget = styled.div`
  background: ${colors.white};
  padding: 16px;
  width: 260px;
`;
StyledWidget.displayName = 'Widget';

const Widget = props => {
  const { children, description, title } = props;

  return (
    <StyledWidget>
      {title && <SubHeader>{title}</SubHeader>}
      {description && <Text color={colors.type02}>{description}</Text>}
      {children}
    </StyledWidget>
  )
};

Widget.defaultProps = {
  title: '',
  description: '',
};

Widget.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
};

export default Widget;
