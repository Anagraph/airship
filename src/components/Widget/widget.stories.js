import React from 'react';
import { storiesOf } from '@storybook/react';
import { ThemeProvider } from 'styled-components';
import Widget from './widget';
import Text from '../Typography/text';
import { theme } from '../../constants';

const CUSTOM_THEME = {
  ...theme,
  type01: '#FFF',
  type02: '#EEE',
  ui01: '#333',
};

storiesOf('Widget', module)
  .add('Default', () => (
    <div>
      <Widget>
        <Widget.Title>Populated Places</Widget.Title>
        <Widget.Description>All selected</Widget.Description>
        <Text>I am the senate!</Text>
      </Widget>
    </div>
  ))
  .add('With custom theme', () => (
    <ThemeProvider theme={CUSTOM_THEME}>
      <Widget>
        <Widget.Title>Populated Places</Widget.Title>
        <Widget.Description>All selected</Widget.Description>
        <Text>I am the senate!</Text>
      </Widget>
    </ThemeProvider>
  ));
