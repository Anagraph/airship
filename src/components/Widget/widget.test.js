import React from 'react';
import Widget from './widget';
import renderer from 'react-test-renderer';

describe('render', () => {
  it('renders without crashing', () => {
    const component = renderer.create(
      <Widget />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with the title prop', () => {
    const component = renderer.create(
      <Widget title="Rick & Morty" />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with the decription prop', () => {
    const component = renderer.create(
      <Widget description="Rick & Morty" />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders children', () => {
    const component = renderer.create(
      <Widget>This is a children</Widget>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
