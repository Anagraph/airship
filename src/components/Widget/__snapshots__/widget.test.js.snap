// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders children 1`] = `
.c0 {
  background: #fff;
  padding: 16px;
  width: 260px;
}

.c1 {
  color: #747474;
  margin: 0;
  font-family: Roboto,sans-serif;
  font-weight: 400;
  font-size: 12px;
  line-height: 20px;
}

<div
  className="c0"
>
  
  <p
    className="c1"
    color="#747474"
  >
    Rick & Morty
  </p>
  This is a children
</div>
`;

exports[`render renders with the decription prop 1`] = `
.c0 {
  background: #fff;
  padding: 16px;
  width: 260px;
}

.c1 {
  color: #747474;
  margin: 0;
  font-family: Roboto,sans-serif;
  font-weight: 400;
  font-size: 12px;
  line-height: 20px;
}

<div
  className="c0"
>
  
  <p
    className="c1"
    color="#747474"
  >
    Rick & Morty
  </p>
</div>
`;

exports[`render renders with the title prop 1`] = `
.c0 {
  background: #fff;
  padding: 16px;
  width: 260px;
}

.c1 {
  margin: 0;
  font-family: Roboto,sans-serif;
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Rick & Morty
  </h1>
  
</div>
`;

exports[`render renders without crashing 1`] = `
.c0 {
  background: #fff;
  padding: 16px;
  width: 260px;
}

<div
  className="c0"
>
  
  
</div>
`;
